File: redsheet-portal/src/app/database-models/lite-project-background.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { LiteRelationshipRequire, ProjectModule, LiteBackgroundCommunicationMode, LiteTheirTeamMember, LiteTarget, LiteOurTeamMember,  vLiteProjectBackgroundRecord  } from './'
export class LiteProjectBackground {
    constructor(liteProjectBackground?: LiteProjectBackground  | vLiteProjectBackgroundRecord )  {
        let properties = [ "dateOfNegotiation", "focus", "knowAboutThem", "knownIssues", "liteProjectBackgroundId", "location", "opponentName", "reason", "valueObjectiveId", "liteRelationshipRequireId", "projectModuleId", "liteBackgroundCommunicationModes", "liteTheirTeamMembers", "liteTargets", "liteOurTeamMembers",];
        for (let property of properties)
            if (liteProjectBackground && liteProjectBackground[property])
                this[property] = liteProjectBackground[property];
    }

    @required()
	dateOfNegotiation : Date =   undefined;

    @required()
    @maxLength(200)
	focus : string =   undefined;

    @required()
    @maxLength(400)
	knowAboutThem : string =   undefined;

    @required()
    @maxLength(400)
	knownIssues : string =   undefined;

	liteProjectBackgroundId : number =   0 ;

    @required()
    @maxLength(50)
	location : string =   undefined;

    @required()
    @maxLength(100)
	opponentName : string =   undefined;

    @required()
    @maxLength(200)
	reason : string =   undefined;

    @range(1,2147483647)
	valueObjectiveId : number =   undefined;

    @range(0,2147483647)
	liteRelationshipRequireId : number =   undefined;
	liteRelationshipRequire : LiteRelationshipRequire  ;

    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
	@nested(LiteBackgroundCommunicationMode)
	liteBackgroundCommunicationModes: LiteBackgroundCommunicationMode[];

	@nested(LiteTheirTeamMember)
	liteTheirTeamMembers: LiteTheirTeamMember[];

	@nested(LiteTarget)
	liteTargets: LiteTarget[];

	@nested(LiteOurTeamMember)
	liteOurTeamMembers: LiteOurTeamMember[];



}

File: RedSheet.DbEntities/Models/LiteProjectBackground.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("LiteProjectBackgrounds", Schema = "dbo")]
    [Serializable]
    public partial class LiteProjectBackground
    {

        [Required]
        public System.DateTime DateOfNegotiation { get; set; }

        [Required]
        [MaxLength(200)]
        public string Focus { get; set; }

        [Required]
        [MaxLength(400)]
        public string KnowAboutThem { get; set; }

        [Required]
        [MaxLength(400)]
        public string KnownIssues { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int LiteProjectBackgroundId { get; set; }

        [Required]
        [MaxLength(50)]
        public string Location { get; set; }

        [Required]
        [MaxLength(100)]
        public string OpponentName { get; set; }

        [Required]
        [MaxLength(200)]
        public string Reason { get; set; }

        [Range(1, int.MaxValue)]
        public int ValueObjectiveId { get; set; }

        [RelationshipTableAttribue("LiteRelationshipRequires", "dbo")]
        //Relationships
        public int LiteRelationshipRequireId { get; set; }

        [ForeignKey("LiteRelationshipRequireId")]
        public LiteRelationshipRequire LiteRelationshipRequire { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }
        [InverseProperty("LiteProjectBackground")]
        public ICollection<LiteBackgroundCommunicationMode> LiteBackgroundCommunicationModes { get; set; }

        [InverseProperty("LiteProjectBackground")]
        public ICollection<LiteTheirTeamMember> LiteTheirTeamMembers { get; set; }

        [InverseProperty("LiteProjectBackground")]
        public ICollection<LiteTarget> LiteTargets { get; set; }

        [InverseProperty("LiteProjectBackground")]
        public ICollection<LiteOurTeamMember> LiteOurTeamMembers { get; set; }


        public LiteProjectBackground()
        {
            this.LiteBackgroundCommunicationModes = new HashSet<LiteBackgroundCommunicationMode>();

            this.LiteTheirTeamMembers = new HashSet<LiteTheirTeamMember>();

            this.LiteTargets = new HashSet<LiteTarget>();

            this.LiteOurTeamMembers = new HashSet<LiteOurTeamMember>();

        }

    }
}

